### PR TITLE
DAOS-1457 Tests: Add nvme tag to avocado tests that are worth running on SSDs

### PR DESCRIPTION
**Pull request by omar-ocampo**
Added nvme,pmem and performance tags to the appropiate tests
based on their yaml configuration.

Signed-off-by: Omar Ocampo <omar.ocampo.coronado@intel.com>
Features: nvme pmem performance

### CODE REVIEW

**Comment by omar-ocampo** [comment]
To simplify the reviews, the ticket is split into a couple of PRs

File: src/tests/ftest/control/daos_admin_privileged.py [hunk 1]
```diff
@@ -31,7 +31,10 @@ def test_daos_admin_format(self):
             Test daos_admin functionality to perform format privileged
             operations while daos_server is run as normal user.
 
-        :avocado: tags=all,pr,daily_regression,hw,small,daos_admin,basic
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,small
+        :avocado: tags=pmem
```
Comment (phender) [inline src/tests/ftest/control/daos_admin_privileged.py:36]: Based on https://github.com/daos-stack/daos/blob/master/src/tests/ftest/control/daos_admin_privileged.yaml#L18-L20 this test should be using NVMe.  I wonder if it does not do any harm to add the tag now and then when the comment is removed in the yaml file it does not end up being forgotten.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/control/daos_admin_privileged.py (hunk 1 above)
Comment (omar-ocampo) [reply src/tests/ftest/control/daos_admin_privileged.py:36]: DAOS-4287, which is the reason why the nvme part is commented out in the yaml is already marked as resolved, I can try to uncomment the code and add the nvme tag

**Comment by brianjmurrell** [comment]
You could take advantage of the `Quick-Functional: true` pragama here to build quickly and cut right to Functional testing.

**Review by brianjmurrell — COMMENTED** [review]
While I'm all for grooming the tags in our tests to better enable developers to target `Features:`, doesn't a broad tag such as `nvme` or `pmem` just end up including most tests?

Is this (seemingly -- maybe it's not really as large as it seems) large number of tests all of the tests that we have that use nvme and/or pmem or is this really only a smaller subset that is _worth running_ (to use the words from the ticket)?

**Comment by phender** [comment]
> While I'm all for grooming the tags in our tests to better enable developers to target `Features:`, doesn't a broad tag such as `nvme` or `pmem` just end up including most tests?

By themselves, that statement would be mostly accurate (assuming they are applied to all tests that use `scm_class: dcpm` and `bdev_class: nvme` test yaml entries.

> Is this (seemingly -- maybe it's not really as large as it seems) large number of tests all of the tests that we have that use nvme and/or pmem or is this really only a smaller subset that is _worth running_ (to use the words from the ticket)?

I would argue `nvme` is not the correct tag to deem a test 'worthy' of running.  Tags like `basic`, `coverage`, `release_testing`, `test_stage_N`, etc. are better used to defined when it is the correct time to run the test - i.e. the execution of the test is worth while.

**Comment by brianjmurrell** [comment]
> I would argue `nvme` is not the correct tag to deem a test 'worthy' of running. Tags like `basic`, `coverage`, `release_testing`, `test_stage_N`, etc. are better used to defined when it is the correct time to run the test - i.e. the execution of the test is worth while.

While I generally agree, and even as old as the ticket is (I think it by far predates the `Features:` pragma) I think the spirit of the ticket is **exactly** what `Features:` is intended to do.  In that spirit, I think the ticket owner is looking to tag _some subset_ of tests, (much?) smaller than the full set, that represent a good, broad, over-all coverage of testing functionality for NVMe devices.  Indeed, it really should be a very thoughtfully selected set of tests and not just some handful chosen at more or less random, and they should try to strive for broad coverage of functionality, not lots of overlapping functionality.  IOW, just because a test is configured to use NVMe hardly makes it a _worthy_ candidate.  Just about every test is going to use NVMe as that's a core component of DAOS.

I would not necessarily disagree that perhaps such a generic tag as `nvme` is perhaps the wrong tag/model for `Features:`, and that it really maybe should be applied to every test that uses NVMe, I'm not sure what the tag should be to represent the _worthy_ set.

**Comment by omar-ocampo** [comment]
I really don't have any strong opinion regarding what should be tagged with nvme/pmem or not. There are different paths and each of them is valid. The same conversation would be in place with other tags.

Seems we have two options: 

1. Tag with nvme if the test uses nvme
2. Tag with nvme only if the test is testing the nvme feature (nvme folder only?)

Please let me know your preference so I can adapt this PR.

Cheers

**Comment by brianjmurrell** [comment]
> I really don't have any strong opinion regarding what should be tagged with nvme/pmem or not.

But I think the description in the ticket does.  In my understanding of the ticket, the goal is not to tag every single test that uses NVMe devices with `nvme` otherwise somebody using `Features: nvme` because they are modifying something specific to how NVMe is handled ends up running close to all of the tests, assuming "close to all" use NVMe devices.  My understanding of the ticket is to **carefully** choose a group of tests that exercises some different NVMe functionalities without a ton of overlap as overlap just wastes testing cycles running the same kinds of tests over and over again.

> There are different paths and each of them is valid. The same conversation would be in place with other tags.

Absolutely.  This conversation/concept applies to all of the tags we are trying to add to represent "Features:`.


> Seems we have two options:
> 
> 1. Tag with nvme if the test uses nvme

And have long test cycles, consuming test resources for unnecessarily long times, testing things completely unrelated to NVMe just because the test, unsurprisingly (given that NVMe is the bulk storage backing for DAOS) uses NVMe devices.

> 2. Tag with nvme only if the test is testing the nvme feature (nvme folder only?)

This is my understanding of the spirit of the ticket.  So this means understanding the various different things that we do with NVMe drives (other than simply having them present so that there is somewhere to store the data that is ancilliary to some actually-non-NVMe test) and finding the tests that exercise those things.

That's not to say that necessarily, there is no place for tagging every single test that uses NVMe with some kind of tag.  I just don't think that is the point of the ticket this PR is supposed to be addressing.

**Pull request closed without merging**